### ISSUE TITLE
NumericUpDown complaining if binding property is NULL

### ISSUE THREAD

**Issue by sekulicb** [MahApps/MahApps.Metro#3786]
Hi. I have specific requirements where I need NumericUpDown control, to hold the value of "start year" which can be any year starting from 1950 but it can also be null, if user doesn't set this field up. My property type is nullable int, since this is the most convenient type. But in the output window I see that this control doesn't like that:

System.Windows.Data Error: 6 : 'SystemConvertConverter' converter failed to convert value '<null>' (type '<null>'); fallback value will be used, if available. BindingExpression:Path=EndYear; DataItem='ServiceDto' (HashCode=6499254); target element is 'NumericUpDown' (Name='EndYear'); target property is 'Value' (type 'Nullable`1') InvalidCastException:'System.InvalidCastException: Null object cannot be converted to a value type.
   at System.Convert.ChangeType(Object value, Type conversionType, IFormatProvider provider)
   at MS.Internal.Data.SystemConvertConverter.Convert(Object o, Type type, Object parameter, CultureInfo culture)
   at System.Windows.Data.BindingExpression.ConvertHelper(IValueConverter converter, Object value, Type targetType, Object parameter, CultureInfo culture)'

These properties are set on the control itself:
Maximum="2050"
Minimum="1950"
NumericInputMode="Numbers"

WPF, .NET Framework 4.7.2
MahApps.Metro 1.6.5

Thanks

**Comment by timunie** [comment]
Hi @sekulicb ,

I think this problem is solved in the current prerelease v2.0 alpha. There are some breaking changes if you want to upgrade, here is a first draft of the release notes: https://github.com/MahApps/MahApps.Metro/wiki/Release-2.0.0

Happy coding
Tim

**Comment by sekulicb** [comment]
Hi @timunie 

Upgrading right now is not an option, because of the delays that it might cause right now, but I manged to solve this problem by placing IValueConverter on the property that is bound to Value property on the NumberUpDown control.

It's a simple hack, but at least I don't get that exception anymore: 

`public static NumericFieldConverter Instance = new NumericFieldConverter();

        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            return value;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            return value;
        }`

I'm gonna leave it here just in case somebody also has the same problem. Closing. Thanks
